functions/graphql/schema/badge.js

```diff
@@ -59,91 +70,101 @@ module.exports.schema = gql`
-      const badge = await Badge.getBadge(badgeId);
-      if (badge.centerId !== center.id)
-        throw new ForbiddenError("Cannot delete a badge you do not own.");
-      return badge.delete();
     },
-    badgeRename: async (rootQuery, { badgeId, name }, context) => {
-      const center = await getCenter(context.user);
-      const badge = await Badge.getBadge(badgeId);
-      if (badge.centerId !== center.id)
-        throw new ForbiddenError("Cannot edit a badge you do not own.");
-      return badge.rename(name);
-    },
-    badgeChangeDescription: async (
-      rootQuery,
-      { badgeId, description },
-      context
-    ) => {
-      const center = await getCenter(context.user);
-      const badge = await Badge.getBadge(badgeId);
-      if (badge.centerId !== center.id)
-        throw new ForbiddenError("Cannot edit a badge you do not own.");
-      return badge.changeDescription(description);
+    Mutation: {
+        badgeCreate: async (rootQuery, { badge }, context) => {
+            const center = await getCenter(context.user);
+            return Badge.createBadge(badge, center.id);
+        },
+        badgeRemove: async (rootQuery, { badgeId }, context) => {
+            const center = await getCenter(context.user);
+            const badge = await Badge.getBadge(badgeId);
+            if (badge.centerId !== center.id)
+                throw new ForbiddenError("Cannot delete a badge you do not own.");
+            return badge.delete();
+        },
+        badgeRename: async (rootQuery, { badgeId, name }, context) => {
+            const center = await getCenter(context.user);
+            const badge = await Badge.getBadge(badgeId);
+            if (badge.centerId !== center.id)
+                throw new ForbiddenError("Cannot edit a badge you do not own.");
+            return badge.rename(name);
+        },
+        badgeChangeDescription: async (
+            rootQuery,
+            { badgeId, description },
+            context
+        ) => {
+            const center = await getCenter(context.user);
+            const badge = await Badge.getBadge(badgeId);
+            if (badge.centerId !== center.id)
+                throw new ForbiddenError("Cannot edit a badge you do not own.");
+            return badge.changeDescription(description);
+        },
+        badgeChangeImage: async (rootQuery, { badgeId, image }, context) => {
+            const center = await getCenter(context.user);
+            const badge = await Badge.getBadge(badgeId);
+            if (badge.centerId !== center.id)
+                throw new ForbiddenError("Cannot edit a badge you do not own.");
+            return badge.changeImage(image);
+        },
+        badgeAssign: async (rootQuery, { badgeId, flightId, userId }, context) => {
+            // Either assign the badge directly, or create an assignment object
+            let user = null;
+            let badge = Badge.getBadge(badgeId);
```

Comment:
This needs to be `await`-ed
```suggestion
            let badge = await Badge.getBadge(badgeId);
```